explainability-core/src/main/java/org/kie/trustyai/metrics/language/rouge/ROUGE.java

```diff
@@ -0,0 +1,163 @@
+
+import org.kie.trustyai.metrics.language.AbstractNLPPerformanceMetric;
+import org.kie.trustyai.metrics.language.utils.NGramUtils;
+import org.kie.trustyai.metrics.language.utils.SentenceUtils;
+
+public class ROUGE extends AbstractNLPPerformanceMetric<Double, String>{
+
+    private final String rougeType;
+
+    public ROUGE(String rougeType){
+        super();
+        this.rougeType = rougeType;
+    }
+
+    @Override
+    public Double calculate(String reference, String hypothesis) {
+        return calculate(reference, hypothesis, rougeType);
+    }
+
+    /**
+     * Calculates the ROUGE score for a reference and a hypothesis.
+     *
+     * @param reference The reference string.
+     * @param hypothesis The hypothesis string.
+     * @param rougeType The ROUGE type string.
+     * @return The ROUGE score.
+     */
+    public Double calculate(String reference, String hypothesis, String rougeType) {
+        if (rougeType.equals("rougeLsum")){
+            // Split the reference and hypothesis into sentences
+            final String[] referenceSentences = SentenceUtils.splitSentences(reference);
+            final String[] hypothesisSentences = SentenceUtils.splitSentences(hypothesis);
+            return rougeLsum(referenceSentences, hypothesisSentences);
+        }
+        // Remove punctuation from the reference and hypothesis
+        reference = reference.replaceAll("\\p{Punct}", "");
+        hypothesis = hypothesis.replaceAll("\\p{Punct}", "");
+
+        if (rougeType.equals("rouge1") || rougeType.equals("rouge2")){
+            // Tokenize the reference and hypothesis
+            final List<String> hypothesisTokens = Arrays.asList(this.getTokenizer().tokenize(hypothesis));
+            final List<String> referenceTokens =  Arrays.asList(this.getTokenizer().tokenize(reference));
+
+            // Get the ROUGE-N type and calculate the n-grams in the reference and hypothesis accordingly
+            final int n = Character.getNumericValue(rougeType.charAt(5));
+            final Map<String, Integer> referenceNgramCount = NGramUtils.countNgrams(referenceTokens, n);
+            final Map<String, Integer> hypothesisNGramCount = NGramUtils.countNgrams(hypothesisTokens, n);
+            return rougeN(referenceNgramCount, hypothesisNGramCount);
+        }
+        else if (rougeType.equals("rougeL")){
+            // Tokenize the reference and hypothesis
+            final List<String> hypothesisTokens = Arrays.asList(this.getTokenizer().tokenize(hypothesis));
+            final List<String> referenceTokens =  Arrays.asList(this.getTokenizer().tokenize(reference));
+            return rougeL(referenceTokens, hypothesisTokens);
+        }
+        else{
+            return 0.0;
+        }
+    }
+
+    /**
+     * Calculate F1 score given the precison and recall.
+     *
+     * @param precision
+     * @param recall
+     * @return F1 score
+     */
+    public double fMeasure(double precision, double recall){
```

Comment:
This is a core measure that we could reuse in other places. I would put it perhaps as a static method of a class under `trustyai/metrics/utils/F1::calculate(...)` or similar.